src/odemis/gui/model/__init__.py

```diff
@@ -453,24 +466,23 @@ def __init__(self, main):
             tools.add(TOOL_ROA)
             if main.tc_detector:  # Can even show live settings
                 tools.add(TOOL_SPOT)
+                # The SpotConfocalstream, used to control spot mode.
+                # It is set at start-up by the tab controller.
+                self.spotStream = None
 
-        self.tool.choices = tools
```

Comment:
Sorry, I've modified this line, and that caused conflict with your changes. Still, you can write just `self.tool.choices = tools`, and drop the redifinition of `self.tools` below.

Reply:
In addition to putting this line back, you need to drop the line a little below `self.tool = IntEnumerated(TOOL_NONE, choices=tools)`.